### PR TITLE
use SystemExit instead os sys.exit

### LINKED ISSUE #658: Raise exceptions instead of using sys.exit of setup.py
This is not really a bug but using sys.exit in setup.py breaks some automated tools that inspect the package and/or build packages from it.

Would you be open to a PR that changes them to exceptions instead?

### PR DESCRIPTION
**Pull request by ocefpaf**
- [X] Closes #658
 - [ ] Tests added
 - [ ] Fully documented, including `history.rst` for all changes and `api/*.rst` for new API

There are no tests for setup.py as far as I can tell I also think this does not need documentation.
This change should be "seamless" for pyproj but it will allow package parsers to inspect this by catching the BaseException instead of exiting.

Ping @marcelotrevisani who had issues in grayskull when parsing this package.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@snowman2 note that a bare `exception` or `BaseException` would catch the `sys.exit` but this will allow for more elegant code upstream where we can catch `SystemExit` and handle exits differently from other failures.

TL;DR this is not essential to `pyproj` and feel free to close it if you think it is not important here.

**Comment by snowman2** [comment]
Looks like it works fine - I think this is a good change:
```
$ python -m pip install -e .
Obtaining file:///home/snowal/scripts/pyproj
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/snowal/miniconda/bin/python /home/snowal/.local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpahd2zjhh
       cwd: /home/snowal/scripts/pyproj
  Complete output (1 lines):
  ERROR: Minimum supported proj version is 7.1.0, installed version is 4.9.3. For more information see: https://pyproj4.github.io/pyproj/stable/installation.html
  ----------------------------------------
ERROR: Command errored out with exit status 1: /home/snowal/miniconda/bin/python /home/snowal/.local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpahd2zjhh Check the logs for full command output.

```

@ocefpaf do you want to `squash` your commits? Or do you mind if I do a `squash-and-merge`?

**Comment by ocefpaf** [comment]
> @ocefpaf do you want to `squash` your commits? Or do you mind if I do a `squash-and-merge`?

Done.

**Comment by ocefpaf** [comment]
Thanks!

**Comment by snowman2** [comment]
Thanks @ocefpaf!
